crates/rune-macros/src/function.rs

```diff
@@ -208,15 +223,24 @@ impl Function {
         if instance {
             self_type = None;
 
-            name = syn::Expr::Lit(syn::ExprLit {
-                attrs: Vec::new(),
-                lit: syn::Lit::Str(match &attrs.path {
-                    Path::None => name_string.clone(),
-                    Path::Rename(last) | Path::Instance(_, last) => {
-                        syn::LitStr::new(&last.ident.to_string(), last.ident.span())
-                    }
-                }),
-            });
+            name = if let Path::Protocol(protocol) = &attrs.path {
+                syn::Expr::Path(syn::ExprPath {
+                    attrs: Vec::new(),
+                    qself: None,
+                    path: protocol.clone(),
+                })
+            } else {
+                syn::Expr::Lit(syn::ExprLit {
+                    attrs: Vec::new(),
+                    lit: syn::Lit::Str(match &attrs.path {
+                        Path::None => name_string.clone(),
+                        Path::Rename(last) | Path::Instance(_, last) => {
+                            syn::LitStr::new(&last.ident.to_string(), last.ident.span())
+                        }
+                        Path::Protocol(_) => unreachable!(),
```

Comment:
I would prefer some repetition over using `unreachable!`. Building a lit expr is not very hard and I'd be worried about exporting the branch during future refactoring.

Alt something like this is also OK imo:
```rust
name = 'out: {
    let lit = match &attrs.path {
        Path::Protocol(protocol) => break 'out syn::Expr::Path(..),
        Path::None => name_string.clone(),
        Path::Rename(last) | Path::Instance(_, last) => {
            syn::LitStr::new(&last.ident.to_string(), last.ident.span())
        }
    };

    syn::Expr::Lit(..)
}
```